plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -188,6 +195,38 @@ public FileHiveMetastore(NodeVersion nodeVersion, HdfsEnvironment hdfsEnvironmen
         }
     }
 
+    @Override
+    public synchronized long acquireTableExclusiveLock(
+            HiveIdentity identity,
+            AcidTransactionOwner transactionOwner,
+            String queryId,
+            String dbName,
+            String tableName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(dbName, tableName);
+
+        Long existingLockId = tableLockIds.get(schemaTableName);
+        ReentrantLock lock = existingLockId == null ? new ReentrantLock() : tableLocks.get(existingLockId);
+
+        // Wait for current lock holder to release it
+        lock.lock();
+
+        long lockId = new SecureRandom().nextLong();
```

Comment:
Make `Random` a field.

OTOH, why not just a counter?

Reply:
The random number keeps a misbehaving thread from releasing locks it doesn't own, since the release method just takes the lock id it should be hard to guess.

Reply:
But after applying https://github.com/trinodb/trino/pull/12026#discussion_r859746671 it doesn't matter, can use a counter